dronecaster.lua

```diff
@@ -57,6 +66,71 @@ done_init = false
 
 -- init & core
 --------------------------------------------------------------------------------
+
+-- midi
+function build_midi_device_list()
+  midi_devices = {}
+  for i = 1,#midi.vports do
+    local long_name = midi.vports[i].name
+    local short_name = string.len(long_name) > 15 and util.acronym(long_name) or long_name
+    table.insert(midi_devices,i..": "..short_name)
+  end
+end
+
+function midi_event(data)
+  -- global midi setting
+  if midi_enabled == 1 then
```

Comment:
You can flip this condition and return early to reduce nesting, i.e.:
```lua
if midi_enabled == 0 then
  return
end
-- ... (previous if-body) ...
```

Reply:
Thanks, will change.

Reply:
Should this be `0` instead of `1` now? Please re-resolve if I'm wrong!

Reply:
Yes, you're right, sorry! Fixed now.